torch/_inductor/pattern_matcher.py

```diff
@@ -711,6 +711,36 @@ def decorator(handler):
     return decorator
 
 
+def is_start_of_fx_graph(node):
+    return len(node.all_input_nodes) == 0
```

Comment:
This doesn't seem right.  Wouldn't this be true for `torch.tensor(0)` or `torch.zeros([])`?

Reply:
I think this depends on what we mean by the "start of the FX graph". I implemented it as "a node that has no inputs," but perhaps there needs to be the additional condition "a node that is before any mutation op"?

Reply:
It should be the literal first node in the graph.